runtime/onert/frontend/base_loader/include/base_loader.h

```diff
@@ -377,18 +377,34 @@ ir::OperandIndex BaseLoader<LoaderDomain>::loadOperand(const Tensor *tensor, ir:
           throw std::runtime_error("traversal_order [0, 1, ..., n-1] is only supported.");
       }
     }
-    // ignore block_map
+    // check block_map
+    int block_rank = 0;
+    if (src_sparsity->block_map())
+    {
+      block_rank = src_sparsity->block_map()->size();
+      for (int i = 0; i < block_rank; ++i)
+      {
+        if (i != src_sparsity->block_map()->Get(i))
+          throw std::runtime_error("block_map [0, 1, ..., n-1] is only supported.");
+      }
+    }
     // load metadata
-    const size_t dim_metadata_size = src_sparsity->dim_metadata()->size();
-    if (dim_metadata_size != 2)
-      throw std::runtime_error("sparse tensor is supported only for 2D");
+    const int dim_metadata_size = src_sparsity->dim_metadata()->size();
+    auto dense_rank = shape.rank();
+    if (dense_rank + block_rank != dim_metadata_size)
+      throw std::runtime_error("sparsity dim_metadata length is wrong.");
+    bool random_sparsity = dim_metadata_size == 2 && block_rank == 0;
+    bool block16x1_sparsity = dim_metadata_size == 4 && block_rank == 2;
```

Comment:
I wonder why dim_metadata_size == 4 && block_rank == 2 conditional is a 16x1 block.

Reply:
@YiHyunJin First two `dim_metadata`s are for 2D block (16x1) and next two `dim_metadata`s are for two dimensions within block. Please see `tflite_schema.fbs` for details.